contracts/SoulBoundToken/BasicSBT.sol

```diff
@@ -0,0 +1,214 @@
+     * The fields within the struct can be be edited for the usecase of the SBT.
+     *
+     * The fields of `identity` and `uri` can be hashed for privacy.
+     */
+    struct Soul {
+        string identity;
+        string uri;
+    }
+
+    // Events
+    event Mint(address _soul);
+    event Burn(address _soul);
+    event Update(address _soul);
+
+    /**
+     * @dev This modifier checks that the address passed in is not the zero address.
+     */
+    modifier validAddress(address _addr) {
+        require(_addr != address(0), "Not valid address");
+        _;
+    }
+
+    /**
+     * @dev Initializes the contract by setting a `name` and a `symbol` to the SBT
+     */
+    constructor(string memory name_, string memory symbol_) {
+        _name = name_;
+        _symbol = symbol_;
+        _totalSBT = 0;
+    }
+
+    /**
+     * @dev Mints `SBT` and transfers it to `_soul`.
+     *
+     * Emits a {Mint} event.
+     */
+    function mint(address _soul, Soul memory _soulData)
+        external
+        virtual
+        validAddress(_soul)
+    {
+        require(!hasSoul(_soul), "Soul already exists");
+        souls[_soul] = _soulData;
+        _totalSBT++;
+        _owners[_totalSBT] = _soul;
+        emit Mint(_soul);
+    }
+
+    /**
+     * @dev Destroys SBT for a given address.
+     *
+     * Requirements:
+     * Only the owner of the SBT can destroy it.
+     * Emits a {Burn} event.
+     */
+    function burn(address _soul) external virtual validAddress(_soul) {
+        require(hasSoul(_soul), "Soul does not exists");
+        require(
+            msg.sender == _soul,
+            "Only users have rights to delete their data"
+        );
+        delete souls[_soul];
+        delete _owners[_totalSBT];
+        emit Burn(_soul);
+    }
+
+    /**
+     * @dev Updates the mapping of address to attribute.
+     * Only the owner address is able to update the information.
+     *
+     * However, projects can have it such that users can propose changes for the contract owner to update.
+     */
+    function updateSBT(address _soul, Soul memory _soulData)
+        public
+        onlyOwner
+        returns (bool)
+    {
+        require(hasSoul(_soul), "Soul does not exist");
+        souls[_soul] = _soulData;
+        emit Update(_soul);
+        return true;
+    }
+
+    /**
+     * @dev Returns the soul data of `identity, uri` for the given address
+     */
+    function getSBTData(address _soul)
+        public
+        view
+        virtual
+        validAddress(_soul)
+        returns (string memory, string memory)
+    {
+        return (souls[_soul].identity, souls[_soul].uri);
+    }
+
+    /**
+     * @dev Validates if the _soul is associated with the valid data in the corresponding address.
+     * By checking if the `_soulData` given in parameter is the same as the data in the struct of mapping.
+     */
+    function validateAttribute(address _soul, Soul memory _soulData)
```

Comment:
I realised that we might not even need this function as well. Since projects can already get the data, they can do the verification on chain. 

🤔 thinking if there would be a reason why projects would want a onchain verification function? 

any thoughts regarding this?

Reply:
You mean that they can do the verification off chain with `getSbtData`? They may not want to write code to verify off chain? 🤔 

But yes without the code will be really short and simple - more introduction and thoughts rather than implementation

Reply:
yup they can do the verification off chain with getSbtData.

Ill leave it in for the discretion of the projects.

Since the function is marked `virtual` the project inheriting this code can replace the implementation.

Ill leave this in as a general idea, as it is a key idea of SBT. Without it its not really sbt.

Reply:
Add a comment as well to explain that it is for their discretion, could also match with the long form article